lib/type1.g

```diff
@@ -231,6 +221,7 @@ BIND_GLOBAL( "NEW_TYPE", function ( typeOfTypes, family, flags, data )
         cached := cache[hash];
         if IS_EQUAL_FLAGS( flags, cached![2] )  then
             if    IS_IDENTICAL_OBJ(  data,  cached![ POS_DATA_TYPE ] )
+              and IS_IDENTICAL_OBJ( parent, cached![ POS_PARENT_TYPE] )
```

Comment:
Answering just (1), as I checked, the test does fail occasionally, but that seemed to be because there were types which were the same, but not IS_IDENTICAL_OBJ. Unfortunately types can't be compared with =, so we can't check this case (and I didn't dive in deeply enough).

To your other general comments, they may be a better way of doing all this.